core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -154,6 +154,13 @@ private void assertNotDot(GenericLexer lexer, CharSequence tok) throws SqlExcept
         }
     }
 
+    //prevent full/right from being used as table aliases
+    private void checkSupportedJoinType(GenericLexer lexer, CharSequence tok) throws SqlException {
+        if (tok != null && (Chars.equalsIgnoreCase(tok, "full") || Chars.equalsIgnoreCase(tok, "right"))) {
```

Comment:
we should have `SqlKeywords` for these comparisons

Reply:
Fixed